fem/bilininteg.hpp

```diff
@@ -2652,11 +2660,23 @@ class ElasticityIntegrator : public BilinearFormIntegrator
 /** Integrator for the DG form:
     alpha < rho_u (u.n) {v},[w] > + beta < rho_u |u.n| [v],[w] >,
     where v and w are the trial and test variables, respectively, and rho/u are
-    given scalar/vector coefficients. The vector coefficient, u, is assumed to
-    be continuous across the faces and when given the scalar coefficient, rho,
-    is assumed to be discontinuous. The integrator uses the upwind value of rho,
-    rho_u, which is value from the side into which the vector coefficient, u,
-    points. */
+    given scalar/vector coefficients. {v} represents the average value of v on
+    the face and [v] is the jump such that {v}=(v1+v2)/2 and [v]=(v1-v2) for the
+    face between elements 1 and 2. For boundary elements, v2=0. The vector
```

Comment:
It's not only `v2=0`, it's `{v}=v1` (there is no division by 2) and `[v]=v1`.

Reply:
The original confusion was that the division by 2 *is* applied for boundary and internal elements alike.
I'm not sure what the *correct* answer is but my main goal was to document how the integrator is truly applied.

Alternatively, the integrator could be modified to not divide by 2 on boundary elements but this would change the current behavior of the code.

Reply:
@YohannDudouit, we should double check this, but I think for this integrator we do use `{v}=(1/2)*v1` on the boundary. When combined with the jump term `[v]=v1` which is scaled by `1/2`, this gives the "upwind" value of `v` on outflow boundary, or `0` on the inflow boundary, see https://github.com/mfem/mfem/issues/1896#issuecomment-736190325.

Reply:
Well... Shouldn't rather fix the code instead of documenting accurately a problem in the code?
```c++
a = (ndof2?0.5:1) * alpha * un;
```

Reply:
I don't think this is a problem in the code. I think this is the right formulation. Why do you think this is a problem?

Reply:
Yes, I agree with Veselin that the code is correct as it is. I would say that it is "more common" in papers to see the notation `{v} = v1` on the boundary rather than `{v} = 0.5*v1`, but in the end it's a notational convention. Since this is spelled out in the comment, I think it's ok.

The important thing is that with the parameters given as specified, the resulting discretization should be the standard upwind DG discretization, which is the case.

Reply:
I remember arguing a lot with my PhD supervisor that `{v} = 0.5*v1` worked for Dirichlet BC and he insisted that it was wrong... I can't remember exactly his argument, but I think that basically you loose a bit on the convergence rate.